storm-client/src/jvm/org/apache/storm/daemon/worker/WorkerState.java

```diff
@@ -717,11 +718,19 @@ private Assignment getLocalAssignment(IStormClusterState stormClusterState, Stri
 
         IWaitStrategy backPressureWaitStrategy = IWaitStrategy.createBackPressureWaitStrategy(topologyConf);
         Map<List<Long>, JCQueue> receiveQueueMap = new HashMap<>();
+
         for (List<Long> executor : executors) {
             List<Integer> taskIds = StormCommon.executorIdToTasks(executor);
+            int taskId = taskIds.get(0);
+            String compId;
+            if (taskId == Constants.SYSTEM_TASK_ID) {
+                compId = Constants.SYSTEM_COMPONENT_ID;
```

Comment:
Should StormCommon.stormTaskInfo() be addressing this mapping?

Reply:
StormCommon.stormTaskInfoImpl() address taskID starting from 1. 
WorkerState.readWorkerExecutors() additionally add Constants.SYSTEM_EXECUTOR_ID (-1) before reading from assignment. This system executor is not present in `taskToComponent` map.